lib/node_modules/@stdlib/lapack/base/dgebak/lib/base.js

```diff
@@ -0,0 +1,104 @@
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var dscal = require( '@stdlib/blas/base/dscal' ).ndarray;
+var swaps = require( './swaps.js' );
+
+
+// MAIN //
+
+/**
+* Forms the right or left eigenvectors of a real general matrix by backward transformation on the computed eigenvectors of the balanced matrix output by `dgebal`.
+*
+* ## Notes
+*
+* The job parameter can be one of the following:
+*
+* -   `none`: none, return immediately
+* -   `permute`: permute only
+* -   `scale`: scale only
+* -   `both`: both permute and scale
+*
+* @private
+* @param {string} job - type of transformation.
+* @param {string} side - specifies the side of the eigenvectors in `V` (`right` or `left`).
+* @param {NonNegativeInteger} N - number of rows in matrix `V`
+* @param {NonNegativeInteger} M - number of columns in matrix `V`
+* @param {NonNegativeInteger} ilo - index of leftmost column of the submatrix computed by `dgebal` (zero-based)
+* @param {NonNegativeInteger} ihi - index of rightmost column of the submatrix computed by `dgebal` (zero-based)
+* @param {Float64Array} scale - contains the details of the permutations and scaling factors applied when balancing `V`, computed by `dgebal`
+* @param {integer} strideScale - stride of `scale`
+* @param {NonNegativeInteger} offsetScale - starting index for `scale`
+* @param {Float64Array} V - input matrix
+* @param {integer} strideV1 - stride of the first dimension of `V`
+* @param {integer} strideV2 - stride of the second dimension of `V`
+* @param {NonNegativeInteger} offsetV - starting index for `V`
+* @returns {Float64Array} `V`
+*
+* @example
+* var Float64Array = require( '@stdlib/array/float64' );
+*
+* var scale = new Float64Array( [ 0.5, 2.0, 1.0 ] );
+* var V = new Float64Array( [ 1.0, 4.0, 2.0, 5.0, 3.0, 6.0 ] );
+*
+* var out = dgebak( 'scale', 'right', 3, 2, 0, 2, scale, 1, 0, V, 2, 1, 0 );
+* // returns <Float64Array>[ 0.5, 2, 4, 10, 3, 6 ]
+*/
+function dgebak( job, side, N, M, ilo, ihi, scale, strideScale, offsetScale, V, strideV1, strideV2, offsetV ) { // eslint-disable-line max-len, max-params
+	var i;
+	var s;
+
+	if ( N === 0 || M === 0 || job === 'none' ) {
+		return V;
+	}
+
+	if ( ilo === ihi ) {
+		if ( job === 'permute' || job === 'both' ) {
+			swaps( N, M, ilo, ihi, scale, strideScale, offsetScale, V, strideV1, strideV2, offsetV ); // eslint-disable-line max-len
+		}
+		return V;
+	}
+
+	if ( job === 'scale' || job === 'both' ) {
+		if ( side === 'right' ) {
+			for ( i = ilo; i <= ihi; i++ ) {
+				s = scale[ offsetScale + ( i * strideScale ) ];
+				dscal( M, s, V, strideV2, offsetV + ( i * strideV1 ) );
+			}
```

Comment:
```suggestion
			is = offsetScale + ( ilo*strideScale );
			for ( i = ilo; i <= ihi; i++ ) {
				s = scale[ is ];
				dscal( M, s, V, strideV2, offsetV + ( i * strideV1 ) );
				is += strideScale;
			}
```

Also applies to offsetV.